From f8d2f35a7161e80866319dc012e876e5d682739c Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 23 Dec 2020 20:04:13 +0200
Subject: [PATCH] [SW-20154] make warm reset configuration platform specific

Configuration of PRSTN to reset the ARCs/PSOC during
warm reset is platform specific. Because, in Gaudi2,
it involves programming the PSOC RESET registers, which
are different than Goya2. Hence move them to platform
specific code

Change-Id: I00fa6d8c0ff0e07e02f249dc97433f34e741ddf6
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 3210e2b..6465394 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -107,6 +107,8 @@
 	}
 }
 
+__attribute__((weak)) void update_warm_reset_config(void){}
+
 static void arcboot_rom_init(void)
 {
 	uint32_t reg;
@@ -123,10 +125,7 @@
 	regmap_writel(0, mmPSOC_GLOBAL_CONF_PRSTN_MASK);
 
 	/* on warm reset change default to reset ARC and PSOC */
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
-	regmap_writel(reg, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	update_warm_reset_config();
 
 	/* read from cold reset flops the recovery and validation flag */
 	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
diff --git a/common/common.h b/common/common.h
index 812b9b1..9745a39 100644
--- a/common/common.h
+++ b/common/common.h
@@ -10,5 +10,6 @@
 uint8_t get_sris_mode_from_strap(void);
 void enable_flr_handling_fsm(void);
 uint32_t get_fld_size_shift(void);
+void update_warm_reset_config(void);
 
 #endif /* _COMMON_H_ */
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 7f230ad..6109d26 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -26,6 +26,15 @@
 	return PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT;
 }
 
+void update_warm_reset_config(void)
+{
+	uint32_t reg;
+	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
+	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
+	regmap_writel(reg, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+}
+
 uint8_t get_sris_mode_from_strap(void)
 {
 	uint32_t boot_strap;
